phygnn/layers/custom_layers.py

```diff
@@ -1473,52 +1527,10 @@ def build(self, input_shape):
         """
         self.rank = len(input_shape)
 
-        msg = 'Sup3rConcatEmbeddedObs layer can only accept 4D or 5D data'
-        assert self.rank in (4, 5), msg
-
-        if self.rank == 4:
-            conv_class = tf.keras.layers.Conv2D
-        else:
-            conv_class = tf.keras.layers.Conv3D
-        self.embed_net = []
-        for _, filters in enumerate(self.filters):
-            self.embed_net += [
-                conv_class(
-                    filters=filters,
-                    kernel_size=1,
-                    padding='same'),
-                tf.keras.layers.LeakyReLU(alpha=0.2)
-            ]
-
-    @staticmethod
-    def nan_fill(hi_res_feature):
-        """Fill NaN values in the input tensor with the mean of the non-NaN.
-        If all values are NaN, fill with 0.
-
-        Returns
-        -------
-        hi_res_feature : tf.Tensor
-            Input tensor with NaN values filled with the mean of the non-NaN
-            values or 0 if all values are NaN.
-        mask : tf.Tensor
-            Mask of the input tensor where 1 is not NaN and 0 is NaN.
-        """
-        not_nan = tf.math.logical_not(tf.math.is_nan(hi_res_feature))
-        not_nan_float = tf.cast(not_nan, hi_res_feature.dtype)
-
-        hi_res_zeroed = tf.where(
-            not_nan, hi_res_feature, tf.zeros_like(hi_res_feature))
-
-        count = tf.reduce_sum(not_nan_float)
-        total = tf.reduce_sum(hi_res_zeroed)
-        mean = tf.math.divide_no_nan(total, count)
-
-        return tf.where(not_nan, hi_res_feature, mean), not_nan_float
-
-    def call(self, x, hi_res_feature=None):
-        """Apply the embed net to x, hi_res_feature, and the mask
-        representing where hi_res_feature is not nan. Concatenate the output
-        with x
+    def call(self, x, hi_res_feature=None, exo_data=None):
+        """Apply the embed net to x, hi_res_feature, exogenous data, and the
+        mask representing where hi_res_feature is not nan. Concatenate the
+        output with x
```

Comment:
The embed net isn't really applied to x, is it? I think the only case where x is run through the embed net is if `fill_method=None` which doesnt actually seem to be supported.

Reply:
No it's not, this is a left over from a previous setup. Thanks